### PR TITLE
PLAT-487 + PLAT-573: A.Foo -> B.Bar, pulse inc before VCallRequest [B.Bar]

### PR DESCRIPTION
**Pull request by ElSacramento**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/virtual/integration/method_test.go [hunk 1]
```diff
@@ -1728,3 +1732,191 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
...
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			assert.Equal(t, objectAGlobal, res.Callee)
+			assert.Equal(t, []byte("finish A.Foo"), res.ReturnArguments)
+			assert.Equal(t, outgoingRef, res.CallOutgoing)
+
+			assert.False(t, res.DelegationSpec.IsZero())
+			assert.Equal(t, objectAGlobal, res.DelegationSpec.Callee)
+			assert.Equal(t, endPulse, res.DelegationSpec.PulseNumber)
+			assert.Equal(t, outgoingRefGlobal, res.DelegationSpec.Outgoing)
+
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/method_test.go:1882]: there should be token check here and in VDelegatedRequestFinished

Comment (ElSacramento) [reply ledger-core/virtual/integration/method_test.go:1882]: what do you mean? check other fields of `res.DelegationSpec`?

Comment (sexyhamster) [reply ledger-core/virtual/integration/method_test.go:1882]: my bad, misread the code. Why not create token outside and compare it in one go?

Comment (ElSacramento) [reply ledger-core/virtual/integration/method_test.go:1882]: I don't want to check all fields

File: ledger-core/virtual/integration/method_test.go [hunk 2]
```diff
@@ -1728,3 +1732,191 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
...
+	// add checks to typedChecker
+	{
+		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+			assert.True(t, report.DelegationSpec.IsZero())
+
+			switch report.Object {
+			case objectAGlobal:
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Equal(t, int32(1), report.UnorderedPendingCount)
+			case objectBGlobal:
```
Comment (ruz) [inline ledger-core/virtual/integration/method_test.go:1821]: no need in B

File: ledger-core/virtual/integration/method_test.go [hunk 3]
```diff
@@ -1728,3 +1732,191 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
...
+		outgoingCallReason = gen.UniqueGlobalRef()
+
+		startPulse = server.GetPulse().PulseNumber
+		endPulse   pulse.Number
+	)
+
+	// create objects
+	{
+		Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+		Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
```
Comment (ruz) [inline ledger-core/virtual/integration/method_test.go:1779]: no need in B

**Pull request closed without merging**